### PR TITLE
refactor(Core): apply clang-tidy modernize-raw-string-literal

### PR DESCRIPTION
**Pull request by FrancescoBorzi**
- Part of https://github.com/azerothcore/azerothcore-wotlk/issues/3820
- Apply `clang-tidy` with `modernize-raw-string-literal`
https://clang.llvm.org/extra/clang-tidy/checks/modernize-raw-string-literal.html

### How to test the changes

- Build with `-DTOOLS=1`
- [Extract the client files](https://www.azerothcore.org/wiki/Extract-Client-Data)
- Run the server
- Access the game
- All good so far? Then this should be good to go

## How to test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

### CODE REVIEW

**Review by Winfidonarleyan — APPROVED** [review]

**Comment by gengarshadowball** [comment]
Still extracting, however i found this weird situation when extracting vmaps:
https://imgur.com/a/GVQrkr3

File: src/tools/vmap4_extractor/vmapexport.cpp [hunk 1]
```diff
@@ -227,7 +227,7 @@ void ParsMapFiles()
     for (unsigned int i = 0; i < map_count; ++i)
     {
         sprintf(id, "%03u", map_ids[i].id);
-        sprintf(fn, "World\\Maps\\%s\\%s.wdt", map_ids[i].name, map_ids[i].name);
+        sprintf(fn, R"(World\Maps\%s\%s.wdt)", map_ids[i].name, map_ids[i].name);
```
Comment (gengarshadowball) [inline src/tools/vmap4_extractor/vmapexport.cpp:230]: To resolve the build error you should swap sprintf with sprintf_s:

sprintf_s(fn, 512, R"(World\Maps\%s\%s.wdt)", map_ids[i].name, map_ids[i].name);

[you can find more information about it on](https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/sprintf-s-sprintf-s-l-swprintf-s-swprintf-s-l?view=msvc-160)

File: src/tools/vmap4_extractor/wdtfile.cpp [hunk 2]
```diff
@@ -113,6 +113,6 @@ ADTFile* WDTFile::GetMap(int x, int z)
 
     char name[512];
 
-    sprintf(name, "World\\Maps\\%s\\%s_%d_%d.adt", filename.c_str(), filename.c_str(), x, z);
+    sprintf(name, R"(World\Maps\%s\%s_%d_%d.adt)", filename.c_str(), filename.c_str(), x, z);
```
Comment (gengarshadowball) [inline src/tools/vmap4_extractor/wdtfile.cpp:116]: The same should be done here. Codefactor is afraid that the buffer might overflow:
_Does not check for buffer overflows (CWE-120)._
[Top 25 most dangerous software weaknesses](https://cwe.mitre.org/data/definitions/120.html)

**Comment by nicoaravena** [comment]
Testing results:
Compilation done
imagen

Server start without problems
imagen

Game running without problems
imagen

**Review by Si1ker — APPROVED** [review]

**Pull request merged**